src/hooks/useChannelStateListener.ts

```diff
@@ -0,0 +1,54 @@
+import { Types } from 'ably';
+import { ChannelParameters } from '../AblyReactHooks';
+import { useAbly } from './useAbly';
+import { useEventListener } from './useEventListener';
+
+type ChannelStateListener = (stateChange: Types.ChannelStateChange) => any;
+
+export function useChannelStateListener(
+    channelNameOrNameAndOptions: ChannelParameters,
+    listener?: ChannelStateListener
+);
+
+export function useChannelStateListener(
+    channelNameOrNameAndOptions: ChannelParameters,
+    state?: Types.ChannelState | Types.ChannelState[],
+    listener?: ChannelStateListener
+);
+
+export function useChannelStateListener(
+    channelNameOrNameAndOptions: ChannelParameters,
+    stateOrListener?:
+        | Types.ChannelState
+        | Types.ChannelState[]
+        | ChannelStateListener,
+    listener?: (stateChange: Types.ChannelStateChange) => any
+) {
+    const ably = useAbly(
+        typeof channelNameOrNameAndOptions === 'object'
+            ? channelNameOrNameAndOptions.id
+            : undefined
+    );
+
+    const channel =
```

Comment:
It's related to my PR https://github.com/ably-labs/react-hooks/pull/66. Looks like we need to discuss this. Can we simply use `useChannel` to get channel instance?

Reply:
`useChannel` can be used to get a channel instance but all it will do is call `channels.get` itself. The ably-js `channels.get` implementation caches channels internally by name, so I think this is the simplest way to get a handle on a channel instance here

Reply:
The only thing that concern me is that `channels.get` with `options` parameter has a side-effect. Basically it doesn't only get channel but also assigns options to it. And on second invocation it throws an error. That's why if we use `useChannelStateListener` with `options` parameter we will receive an error. May be it will be a good idea just ignore `options` in `useChannelStateListener`?

Reply:
Oh yeah that's a very good point, I agree we should ignore options here 👍 I've updated it now